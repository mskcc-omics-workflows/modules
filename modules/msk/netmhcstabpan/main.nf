process NETMHCSTABPAN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/netmhctools:1.0.0':
        'docker.io/mskcc/netmhctools:1.0.0' }"

    input:
    tuple val(meta), path(inputFasta), val(hlaString), val(inputType)


    output:
    tuple val(output_meta), path("*.netmhcstabpan.output"),   emit: netmhcstabpanoutput
    path "versions.yml",                                      emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def hla = hlaString.trim()
    output_meta = meta.clone()
    output_meta.typeMut = inputType == "MUT" ? true : false
    output_meta.fromStab = true
    """

    /usr/local/bin/netMHCstabpan-1.0/netMHCstabpan -s -1 -f ${inputFasta} -a ${hla} -l 9,10 -inptype 0 > ${prefix}.${inputType}.netmhcstabpan.output

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        netmhcpan: v4.1
        netmhcstabpan: v1.0
    END_VERSIONS

    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_meta = meta.clone()
    output_meta.typeMut = inputType == "MUT" ? true : false
    output_meta.fromStab = true
    """
    touch ${prefix}.MUT.netmhcstabpan.output


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        netmhcpan: v4.1
        netmhcstabpan: v1.0
    END_VERSIONS
    """
}
