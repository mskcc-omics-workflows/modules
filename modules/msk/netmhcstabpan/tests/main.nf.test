nextflow_process {

    name "Test Process NETMHCSTABPAN"
    script "../main.nf"
    process "NETMHCSTABPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "netmhcstabpan"

    test("netmhcstabpan - MUT - xls,output,fa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(file(params.test_data_mskcc['neoantigen']['MUT_sequence_fa']), checkIfExists: true),
                    "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02",
                    "MUT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcstabpanoutput[0][0],
                    file(process.out.netmhcstabpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }

    test("netmhcstabpan - WT - xls,output,fa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(file(params.test_data_mskcc['neoantigen']['WT_sequence_fa']), checkIfExists: true),
                    "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02",
                    "WT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcstabpanoutput[0][0],
                    file(process.out.netmhcstabpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }

    test("netmhcstabpan - xls,output,fa - stub") {

        options "-stub"

        when {
            process {

                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('MUT_sequence_fa'),
                    "HLA",
                    "MUT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcstabpanoutput[0][0],
                    file(process.out.netmhcstabpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }


}
