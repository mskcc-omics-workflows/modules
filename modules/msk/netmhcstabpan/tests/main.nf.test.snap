{
    "netmhcstabpan - MUT - xls,output,fa": {
        "content": [
            [
                "versions.yml:md5,d1b3afd365748a44098c5642fad6c94a"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": true,
                "fromStab": true
            },
            "test.MUT.netmhcstabpan.output"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-13T11:11:05.586612"
    },
    "netmhcstabpan - xls,output,fa - stub": {
        "content": [
            [
                "versions.yml:md5,d1b3afd365748a44098c5642fad6c94a"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": true,
                "fromStab": true
            },
            "test.MUT.netmhcstabpan.output"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-13T11:12:06.099529"
    },
    "netmhcstabpan - WT - xls,output,fa": {
        "content": [
            [
                "versions.yml:md5,d1b3afd365748a44098c5642fad6c94a"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": false,
                "fromStab": true
            },
            "test.WT.netmhcstabpan.output"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-13T11:12:01.701587"
    }
}